### DISCUSSION TITLE
After migration into new Tiptap, it fails to read text styling from previous Tiptap

### DISCUSSION THREAD

**Discussion by Arik14** [ueberdosis/tiptap#1919] [General Questions] [answered]
Hello,

I have unique problem where I set up new editor from new version of Tiptap. But when I use html string from output of previous Tiptap, it seems fail to maintain nested structure of this html and lose some of the styling value. For example,
This is html from previous Tiptap using custom extension.
```
<span style="text-align:center;">
  <span style="font-size: 32pt">
    <span style="color: #FFFFFF">
      <span style="font-family: Oswald">Apps for Kids Grades K-2</span>
    </span>
  </span>
</span>
```
But when I use that html and put it into new Tiptap using `setContent` command, the html become like this
```
<span style="font-family: Oswald; font-size: null">Apps for Kids Grades K-2</span>
```

The style `font-size` become `null`, the `color` become missing. I use custom extension and learn it from text-align custom extension from Tiptap github. I don't know what missing, but it seems like everytime I use nested html like this, the style tends to be missing, even if I don't use custom extension at all. For example if I just use font-family. It seems like behavior of new Tiptap is different than previous version.

All the nested style become inline style or merge (but with missing value)
Or did I miss something that cause Tiptap fail to maintain nested structure html above?

Here I create sandbox https://codesandbox.io/s/wandering-wood-etvm7?file=/src/components/HelloWorld.vue
You can inspect element of the text **Hello world**, then click button Toggle Editor and inspect element again.
You can see how it change.

And here is more detail explanation about my problem.
https://github.com/ueberdosis/tiptap/discussions/1815

**Comment by wkl007** [comment]
I have the same problem, hope to get help.

**Reply by WangHaohui-org** [reply]
I have the same problem as you!plz help me!plz!

**Comment by shynline** [comment]
I have the same problem. any workaround or fix?

**Comment by fpernin-bkg** [comment]
same problem for me, do you have an alternative to keep this kind of span structure ?
This one i want to keep like this :
```
<p>
  <span style="color:yellow; font-size: 120px;">sdsd
    <span style="color:red;font-size:50px">bbb
      <span style="font-size:30px">Lorem ipsum</span>
    </span>
  </span>
</p>
```

But the lib by default transform it like this : 
```
<p>
  <span style="color: yellow; font-size: 120px">sdsd</span>
  <span style="color: red; font-size: 50px">bbb</span>
  <span style="font-size: 30px">Lorem ipsum</span>
</p>
```

**Comment by CarlMenke** [comment]
yeah same issue, the problem is that tiptap doesn't support nested marks i believe

**Comment by deki23** [comment]
This might be related to this https://github.com/ueberdosis/tiptap/discussions/1815 and https://github.com/ueberdosis/tiptap/discussions/5213

**Comment by bdbch** [comment, marked as answer]
This is the same issue you described in #1815 I guess? Nicholas already talked about this currently being a Prosemirror limitation. I myself didn't look into this enough to see where exactly it is stripped so I can't really offer a solution right now, I just think we should move this over to #1815 to not have to many discussions with the same topic open.
